src/statistics.jl

```diff
@@ -327,7 +327,7 @@ function apply_statistic(stat::HistogramStatistic,
             x_min = Gadfly.concrete_minimum(values)
             span = x_max - x_min
             binwidth = span / d
-            bincounts ./= sum(bincounts) * binwidth
+            bincounts = bincounts ./ sum(bincounts) * binwidth
```

Comment:
(In-place, fused vectorized operations will be a key feature in Julia 0.6.  The fact that it catches a type instability is a bonus. :wink:)